src/cdk/overlay/position/flexible-connected-position-strategy.ts

```diff
@@ -542,10 +542,10 @@ export class FlexibleConnectedPositionStrategy implements PositionStrategy {
     }
 
     // How much the overlay would overflow at this position, on each side.
-    let leftOverflow = 0 - x;
-    let rightOverflow = (x + overlay.width) - viewport.width;
-    let topOverflow = 0 - y;
-    let bottomOverflow = (y + overlay.height) - viewport.height;
+    let leftOverflow = viewport.left - x;
+    let rightOverflow = (x + overlay.width) - (viewport.left + viewport.width);
+    let topOverflow = viewport.top - y;
+    let bottomOverflow = (y + overlay.height) - (viewport.top + viewport.height);
```

Comment:
These changes seem to double the viewport scroll position in the calculation since the `x` and `y` already have accounted for it. If you take a look at this screenshot, I have an overlay that's right on the edge of being outside the viewport on the left side, but it still has ~5px to go. These changes calculate that it's already overflowing since I've scrolled the viewport ~50px.

![Angular_Material_-_Google_Chrome_2020-10-29_17-15-25](https://user-images.githubusercontent.com/4450522/97601957-096c7f80-1a0b-11eb-876a-58d4aa78b8eb.png)

Reply:
I will look into that.